gcloud/datastore/_implicit_environ.py

```diff
@@ -4,9 +4,63 @@
 imply the current dataset ID and connection from the enviroment.
 """
 
+import httplib2
+import socket
+
+try:
+    from google.appengine.api import app_identity
+except ImportError:
+    app_identity = None
+
 
 DATASET_ID = None
 """Module global to allow persistent implied dataset ID from enviroment."""
 
 CONNECTION = None
 """Module global to allow persistent implied connection from enviroment."""
+
+
+def app_engine_id():
+    """Gets the App Engine application ID if it can be implied.
+
+    :rtype: string or ``NoneType``
+    :returns: App Engine application ID if running in App Engine,
+              else ``None``.
+    """
+    if app_identity is None:
+        return None
+
+    return app_identity.get_application_id()
```

Comment:
I've been trying to make the `None` checks explicit after a previous conversation we had. Do you recall? Shall I still update this chunk of code?